docs/architecture/databaseUtil.md

```diff
@@ -0,0 +1,51 @@
+# DatabaseUtil
+
+The [DatabaseUtil](../src/utils/databaseUtil.ts) abstracts
+writing and reading to the file system,
+as well as parsing and serializing the game data.
+
+## Load
+
+```mermaid
+sequenceDiagram
+  autonumber
+  participant A as Anything
+  participant D as DatabaseUtil
+  participant F as File System
+
+  A ->> D: load()
+  D ->> F: read()
+  F ->> D: serializeData: string
+
+  D ->> D: parse(serializeData)
+  D ->> D: data: unknown
+
+  alt encountered error
+    D ->> A: data: null
+  else
+    D ->> A: data: unknown
+  end
+```
+
+## Save
+
+```mermaid
+sequenceDiagram
+  autonumber
+  participant A as Anything
+  participant D as GameStateUtil
+  participant F as DatabaseUtil
```

Comment:
Typos: `GameStateUtil` was changed to `DatabaseUtil` and `DatabaseUtil` was changed to `File System` in [this commit](https://github.com/normal-devs/klicker-knight/pull/23/commits/49e6cbc7a5d6e7dafe96b48b30e11a69d4311167)